pyccel/ast/numpyext.py

```diff
@@ -366,9 +372,12 @@ def __init__(self, arg, dtype=None, order='C'):
             raise TypeError('we only accept homogeneous arguments')
 
         # Verify dtype and get precision
-        if dtype is None:
+        if not dtype:
```

Comment:
Why did you change this test? Are you sure that `dtype` can't have a value which could evaluate to `False` ?

Reply:
Can it have a value that can be evaluated as `False` ? i was unaware of that, that would apply to all dtype checks i added.

Reply:
`not dtype` is equivalent to `not bool(dtype)`. Some things return `False` when you do `bool(dtype)` e.g:
- empty lists
- `LiteralFalse`
- empty string?

So I just want to check if this handling is intentional or if a comparison with `None` would be more accurate

Reply:
Now that you mention it, i can see that i would more accurate to use a comparison with `None` and throw an error for the case you mentioned (done by `process_dtype`).

Reply:
Fixed in [bdb80e6](https://github.com/pyccel/pyccel/pull/1166/commits/bdb80e6707ae29560e8ae000d3ceb5394f4e3017).